activerecord/lib/active_record/relation/predicate_builder/association_query_value.rb

```diff
@@ -25,7 +25,10 @@ def queries
         def ids
           case value
           when Relation
-            value.select_values.empty? ? value.select(primary_key) : value
+            relation = value
+            relation = relation.select(primary_key) if relation.select_values.empty?
+            relation = relation.rewhere(primary_type => polymorphic_name) if primary_type
```

Comment:
Why `rewhere` rather than just `where` here?

Reply:
Hey, thanks for the quick response. If we don't use `rewhere` we can end up having duplicated where conditions and the test below will fail. It won't change the behavior, it just generates duplicated conditions.

```ruby
Treasure.where(price_estimates: PriceEstimate.where(estimate_of_type: "Treasure"))
#=> SELECT (...) WHERE "treasures"."id" IN (SELECT (...) FROM "price_estimates" WHERE "price_estimates"."estimate_of_type" = 'Treasure' AND "price_estimates"."estimate_of_type" = 'Treasure')
```

Reply:
https://github.com/rails/rails/pull/48362/files#diff-8083725fa779d54362134c43c404662915037be49d44f05c04e114889ff26c71R55-R57

Reply:
I think it's fine. I prefer that over some condition potentially being ignored.